### PR TITLE
Python sdk sample for workspace connections

### PR DESCRIPTION
**Pull request by singankit**
# PR into Azure/azureml-examples

## Checklist

I have:

- [X] read and followed the contributing guidelines

## Changes

- Python SDK sample for Workspace Connection

fixes #

### CODE REVIEW

**Comment by balapv** [comment]
Should this be a separate sample or should connections be a part of the sdk/python/resources/workspace/workspace.ipynb?

File: sdk/python/resources/connections/connections.ipynb [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    "An Azure Machine Learning workspace is secure and trustworthy way for AML users to store credentials to access external resources from AML\n",
+    "\n",
+    "Workspace Connection can be created using the `WorkspaceConnection` class. This class accepts the following key parameters:\n",
+    "- `name` - Name of connection.\n",
+    "- `target` - The URL or ARM resource ID of the external resource.\n",
+    "- `credentials` - The credentials for authenticating to the external resource.\n",
+    "- `type` - The category of external resource for this connection.\n",
+    "\n",
+    "## 2.1 Create a Workspace Connection\n",
+    "In this example we will create a Workspace Connection to an ACR and use username/password credentials\n",
```
[the remark this answers is not in the document]
Comment (singankit) [reply sdk/python/resources/connections/connections.ipynb]: Have added a link to ACR in azure which is the supported container registry

File: sdk/python/resources/connections/connections.ipynb [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azure.ai.ml.entities import WorkspaceConnection\n",
+    "from azure.ai.ml.entities import UsernamePasswordConfiguration\n",
+    "\n",
+    "credentials = UsernamePasswordConfiguration(\n",
+    "    username=\"<user_name_for_acr>\",\n",
```
[the remark this answers is not in the document]
Comment (singankit) [reply sdk/python/resources/connections/connections.ipynb:109]: Added example for various connection type : git, python_feed and container_registry

**Review by balapv — APPROVED** [review]

File: sdk/python/resources/connections/connections.ipynb [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+    "    password=\"<pwd_for_acr>\"\n",
```
Comment (rastala) [inline sdk/python/resources/connections/connections.ipynb:110]: We should educate users not to hardcode passwords, but to use environment variables instead.

File: sdk/python/resources/connections/connections.ipynb [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azure.ai.ml.entities import WorkspaceConnection\n",
+    "from azure.ai.ml.entities import PatTokenConfiguration\n",
+    "\n",
+    "credentials = PatTokenConfiguration(\n",
+    "    pat=\"<pat_to_git_repo>\"\n",
```
Comment (rastala) [inline sdk/python/resources/connections/connections.ipynb:143]: We should educate users not to hardcode passwords, but to use environment variables instead.

File: sdk/python/resources/connections/connections.ipynb [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azure.ai.ml.entities import WorkspaceConnection\n",
+    "from azure.ai.ml.entities import SasTokenConfiguration\n",
+    "\n",
+    "credentials = SasTokenConfiguration(\n",
+    "    pat=\"<sas_to_access_python_feed>\"\n",
```
Comment (rastala) [inline sdk/python/resources/connections/connections.ipynb:176]: We should educate users not to hardcode passwords, but to use environment variables instead.

**Review by rastala — CHANGES_REQUESTED** [review]

File: sdk/python/resources/connections/connections.ipynb (hunk 5 above)
Comment (singankit) [reply sdk/python/resources/connections/connections.ipynb]: Thanks @rastala  have addressed the comment

File: sdk/python/resources/connections/connections.ipynb (hunk 4 above)
Comment (singankit) [reply sdk/python/resources/connections/connections.ipynb]: Thanks @rastala  have addressed the comment

**Review by rastala — APPROVED** [review]

**Pull request merged**